src/features/dotnetTest.ts

```diff
@@ -142,15 +156,35 @@ export default class TestManager extends AbstractProvider {
             }
         }
 
-        const output = this._getOutputChannel();
         output.appendLine('');
         output.appendLine(`Total tests: ${totalTests}. Passed: ${totalPassed}. Failed: ${totalFailed}. Skipped: ${totalSkipped}`);
         output.appendLine('');
 
         return Promise.resolve();
     }
 
-    private _runDotnetTest(testMethod: string, fileName: string, testFrameworkName: string) {
+    private async _recordRunAndGetFrameworkVersion(fileName: string, testFrameworkName: string) {
+
+        await this._saveDirtyFiles();
+        this._recordRunRequest(testFrameworkName);
+        let projectInfo = await serverUtils.requestProjectInformation(this._server, { FileName: fileName });
+
+        let targetFrameworkVersion: string;
+
+        if (projectInfo.DotNetProject) {
+            targetFrameworkVersion = undefined;
```

Comment:
wat

Reply:
This is the same as was previously being used: https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/features/dotnetTest.ts#L172
 I have just extracted it to a helper

Reply:
@rchande: This is because we never pass the target framework in the legacy case.